www/js/home.js

```diff
@@ -17,79 +17,132 @@ function loadHome() {
+    app.infiniteScroll.destroy(newsTab);
+    newsTab.find('.infinite-scroll-preloader').remove();
+  }
+
+  function loadMoreNews() {
+    $.getJSON(API + '/news?page=' + page)
+      .then(function(resp) {
+        addNews(resp.news);
+        page++;
+        loadingNews = false;
+      });
+  }
+
+  function moreNews() {
+    if (!loadingNews) {
+      if (page <= totalPages) {
+        loadingNews = true;
+        loadMoreNews();
+      } else {
+        destroyNewsInfinite();
+      }
+    }
+  }
+
   function initEventStream(events) {
-    var lastDay = null;
+    let lastDay = null;
 
     if (events.length != 0) {
-      for (var event of events) {
+      for (let event of events) {
         event.start = new Date(event.start);
         event.end = new Date(event.end);
 
-        // Add header on new days. We only download events for one week - checking date is enough
+        // Add header on new days
         if (lastDay != event.start.getDate()) {
           dayHTML = app.templates.dayTitleTemplate({date: event.start});
-          eventTab.append(dayHTML);
+          eventList.append(dayHTML);
+          // Update start date to the start date of the last event + 1 day
+          start.setDate(event.start.getDate() + 1);
         }
 
-        var templateHTML = app.templates.dayTemplate({hasEvents: true,
+        let templateHTML = app.templates.dayTemplate({hasEvents: true,
           events: [event]});
-        eventTab.append(templateHTML);
+        eventList.append(templateHTML);
         lastDay = event.start.getDate();
       }
-    } else {
-      var templateHTML = app.templates.dayTemplate({hasEvents: false});
+    }
+    if (lastEvent) {
+      let templateHTML = app.templates.dayTemplate({hasEvents: false});
       eventTab.append(templateHTML);
     }
   }
 
-  function destroyInfinite() {
-    app.infiniteScroll.destroy(newsTab);
-    newsTab.find('.infinite-scroll-preloader').remove();
+  function destroyEventInfinite() {
+    app.infiniteScroll.destroy(eventTab);
+    eventTab.find('.infinite-scroll-preloader').remove();
   }
 
-  function loadMoreNews() {
-    $.getJSON(API + '/news?page=' + page)
-      .then(function(resp) {
-
-        addNews(resp.news);
-        page++;
-        loadingNews = false;
-      });
-  }
-
-  function moreNews() {
-    if (!loadingNews) {
-      if (page <= totalPages) {
-        loadingNews = true;
-        loadMoreNews();
+  function moreEvents() {
+    if (!loadingEvents) {
+      if (!lastEvent) {
+        loadingEvents = true;
+        loadMoreEvents();
       } else {
-        destroyInfinite();
+        destroyEventInfinite();
       }
     }
   }
 
+  function loadMoreEvents() {
+    $.getJSON(API + '/events?start=' + start.yyyymmdd() + '&end=' + end.yyyymmdd())
+      .then(function(resp) {
+        res = resp.events;
+        // If res is small then there most likely are no more events
+        if (res.length <= 7) {
+          lastEvent = true;
+        } else {
```

Comment:
What happens here?
It looks like a lot of data is thrown away (which means more requests to the server, and that's bad)

Reply:
If we get many events in response (>7), only the upcoming seven will actually be used and the start point will be changed to the last of those seven. So yes, some data is thrown away. The problem I had was that I want to fill the whole screen but I don't know how far in the future I have to look to find seven events. We can discuss a better solution later.